File: test-dest/30_throws.swift

func foo() throws {
  throw java.lang.IllegalStateException("Do not call foo")
}

func bar() throws {
  throw java.lang.IllegalStateException("Do not call bar")
}

func fooReturns() throws -> String {
  throw java.lang.IllegalStateException("Do not call fooReturns")
}

func barReturns() throws -> Int32 {
  throw java.lang.IllegalStateException("Do not call bar")
}

func main(args: [String]) {
  do {
    try foo()
    try bar()
    var x = try fooReturns()
    let y = try barReturns()
  } catch {
    print("Success")
  }
}

main([])

File: test-src/30_throws.kt
import java.util.*

@Throws(java.lang.IllegalStateException::class)
fun foo() {
  throw java.lang.IllegalStateException("Do not call foo")
}

@Throws(java.lang.IllegalStateException::class) fun bar() {
  throw java.lang.IllegalStateException("Do not call bar")
}

@Throws(java.lang.IllegalStateException::class)
fun fooReturns(): String {
  throw java.lang.IllegalStateException("Do not call fooReturns")
}

@Throws(java.lang.IllegalStateException::class) fun barReturns(): Int {
  throw java.lang.IllegalStateException("Do not call bar")
}

fun main(args: Array<String>) {
  try {
    foo()
    bar()
    var x = fooReturns()
    val y = barReturns()
  } catch (e: Exception) {
    println("Success")
  }
}
